### PR TITLE
[ryu] Add new port

### PR DESCRIPTION
**Pull request by kreuzerkrieg**
- What does your PR fix? 
Add ryu port
- Which triplets are supported/not supported?
All except UWP
- Have you updated the CI baseline?
Yes
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

**Comment by kreuzerkrieg** [comment]
Could you please take a look what went wrong, the CI output does not give a clue

**Comment by JackBoosY** [comment]
Please resolve these regressions, see [windows](https://dev.azure.com/vcpkg/public/_build/results?buildId=29691&view=artifacts&type=publishedArtifacts) [linux](https://dev.azure.com/vcpkg/public/_build/results?buildId=29692&view=artifacts&type=publishedArtifacts) [osx](https://dev.azure.com/vcpkg/public/_build/results?buildId=29693&view=artifacts&type=publishedArtifacts) log.

**Comment by kreuzerkrieg** [comment]
> Please resolve these regressions, see [windows](https://dev.azure.com/vcpkg/public/_build/results?buildId=29691&view=artifacts&type=publishedArtifacts) [linux](https://dev.azure.com/vcpkg/public/_build/results?buildId=29692&view=artifacts&type=publishedArtifacts) [osx](https://dev.azure.com/vcpkg/public/_build/results?buildId=29693&view=artifacts&type=publishedArtifacts) log.


Not sure I get it. Looks like there is no `bazel` on OSX. Is it ok? Then what the `vcpkg_find_acquire_program(BAZEL)` for?

As for Win/Lin. 

> -- Downloading https://github.com/ulfjack/ryu/archive/v2.0.tar.gz...
> -- Extracting source C:/vsts/_work/4/s/downloads/ulfjack-ryu-v2.0.tar.gz
> -- Using source at C:/vsts/_work/4/s/buildtrees/ryu/src/v2.0-3429a290df
> CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
>     Command failed: bazel build //ryu
>     Working Directory: C:/vsts/_work/4/s/buildtrees/ryu/src/v2.0-3429a290df
>     See logs for more information:
> 
> Call Stack (most recent call first):
>   ports/ryu/portfile.cmake:12 (vcpkg_execute_build_process)
>   scripts/ports.cmake:90 (include)

Is there failed command out put somewhere?

**Comment by JackBoosY** [comment]
@kreuzerkrieg I've added bazel in OSX.

**Comment by kreuzerkrieg** [comment]
> @kreuzerkrieg I've added bazel in OSX.

Thanks!
Any idea what is the meaning of Win/Lin error message?

**Comment by JackBoosY** [comment]
@kreuzerkrieg 
```
    Command failed: bazel build //ryu
    Working Directory: C:/vsts/_work/4/s/buildtrees/ryu/src/v2.0-3429a290df
```
Make sure the command is correct.

**Comment by dan-shaw** [comment]
@kreuzerkrieg The tensorflow port has an example on how to use bazel in vcpkg. I remember you had to use `///` instead of `//` in `bazel build` on Windows for the targets but I forgot why.

**Comment by kreuzerkrieg** [comment]
> @kreuzerkrieg The tensorflow port has an example on how to use bazel in vcpkg. I remember you had to use `///` instead of `//` in `bazel build` on Windows for the targets but I forgot why.

AFAIR, tensorflow just checks if `bazel` installed, as for build it executes some python script. I will check the `///` vs `//` once I get to Windows machine. However I dont get why it fails on Linux, looks like it is running fine on my Ubuntu machine

**Comment by kreuzerkrieg** [comment]
Nailed it.
Took a clean machine to test the port. It failed the exact way it failed in Azure CI.
The reason the `vcpkg_find_acquire_program(BAZEL)` does not deliver. It does not install bazel on the target machine. Once installed manually the port completed successfully

**Comment by JackBoosY** [comment]
@kreuzerkrieg You should use BAZEL downloaded by vcpkg, please see the [example](https://github.com/microsoft/vcpkg/blob/master/ports/tensorflow-cc/portfile.cmake#L90).

**Review by JackBoosY — APPROVED** [review]

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1544,8 +1544,13 @@ rpclib:x86-windows=ignore
 rpclib:x64-windows-static=ignore
 rttr:arm-uwp=fail
 rttr:x64-uwp=fail
-rxspencer:x64-uwp=fail  
+rxspencer:x64-uwp=fail
```
Comment (dan-shaw) [inline scripts/ci.baseline.txt]: Is this failure related to the port?

**Review by dan-shaw — APPROVED** [review]

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (kreuzerkrieg) [reply scripts/ci.baseline.txt]: Sure it isnt. I guess it is EOL settings of my IDE
will try to reconfigure it and revert the change

Comment (kreuzerkrieg) [reply scripts/ci.baseline.txt]: change reverted

**Comment by dan-shaw** [comment]
LGTM, failures seem unrelated

**Comment by strega-nil** [comment]
Alright, thanks @kreuzerkrieg ! This is a great addition :)

**Pull request merged**